imageregistry/v1/types.go

```diff
@@ -306,6 +306,36 @@ type ImageRegistryConfigStorageIBMCOS struct {
 	ServiceInstanceCRN string `json:"serviceInstanceCRN,omitempty"`
 }
 
+// ImageRegistryConfigStorageOSS holds Alibaba Cloud OSS configuration.
+type ImageRegistryConfigStorageOSS struct {
+	// bucket is the bucket name in which you want to store the registry's
+	// data.
+	// Optional, will be generated if not provided.
+	// +optional
+	Bucket string `json:"bucket,omitempty"`
+	// region is the GCS location in which your bucket exists.
+	// Optional, will be set based on the installed GCS Region.
+	// +optional
+	Region string `json:"region,omitempty"`
+	// regionEndpoint is the endpoint for S3 compatible storage services.
+	// Optional, defaults based on the Region that is provided.
+	// +optional
+	RegionEndpoint string `json:"regionEndpoint,omitempty"`
+	// internal specifies whether the registry use the OSS VPC internal endpoint
+	// Optional, defaults to false. if RegionEndpoint is specified, this config will be ignored
+	// +optional
+	Internal bool `json:"internal,omitempty"`
+	// encrypt specifies whether the registry stores the image in encrypted
+	// format or not.
+	// Optional, defaults to false.
+	// +optional
+	Encrypt bool `json:"encrypt,omitempty"`
+	// keyID is the KMS key ID to use for encryption.
```

Comment:
you need some links to external documentation so someone can chase these and figure out what they mean.

Reply:
How will it be mapped into the registry config? I don't see anything about KMS in Distribution config: https://github.com/docker/docker.github.io/blob/master/registry/storage-drivers/oss.md

Reply:
> you need some links to external documentation so someone can chase these and figure out what they mean.

https://www.alibabacloud.com/help/doc-detail/117914.htm?spm=a2c63.p38356.b99.1075.5c3e56989tiYEz

Reply:
> How will it be mapped into the registry config? I don't see anything about KMS in Distribution config: https://github.com/docker/docker.github.io/blob/master/registry/storage-drivers/oss.md

Sorry, I didn't notice this document at first. I think I may need to remove the KMSID configuration, or update registry config, but updating Registry Config may be a huge effort. Any suggestions?

Reply:
So i will remove KeyID